sdk/src/main/java/com/hedera/hashgraph/sdk/Transaction.java

```diff
@@ -792,16 +794,73 @@ public T freezeWith(@Nullable Client client) {
         return (T) this;
     }
 
-    void buildTransactions(int untilIndex) {
-        for (var i = transactions.size(); i < untilIndex; ++i) {
-            transactions.add(com.hedera.hashgraph.sdk.proto.Transaction.newBuilder()
-                .setSignedTransactionBytes(
-                    signedTransactions.get(i)
-                        .setSigMap(signatures.get(i))
-                        .build()
-                        .toByteString()
-                ).build());
+    void buildAllTransactions() {
+        for (var i = 0; i < signedTransactions.size(); ++i) {
+            buildTransaction(i);
+        }
+    }
+
+    /**
+     * Will build the specific transaction at {@code index} and will fill with `null` for any empty indices before it
+     *
+     * @param index
+     */
+    void buildTransaction(int index) {
+        if (transactions.size() < index) {
```

Comment:
With this change if we were to have an empty list `[]` for `transactions` the size would be `0` and the index would be `0`. This conditional would pass since `index + 1` is `1` and `transaction.size()` is `0`. In the for loop within this conditional 
```java
for (var i = transactions.size(); i <= index; ++i) {
```
`i` would be `0` and `index` would be `0` and since we use the `<=` operator this for loops condition would pass resulting in us adding a `null` as the first `transaction`. 
Also, out of curiosity I made this change locally and ran the tests. The test quickly erred with `INVALID_TRANSACTION_BODY`